src/shared/mesh/MeshBuilder.cpp

```diff
@@ -0,0 +1,162 @@
+#include "MeshBuilder.hpp"
+
+#include "axom/slic.hpp"
+#include "axom/primal.hpp"
+
+namespace shared {
+
+MeshBuilder MeshBuilder::Merged( std::initializer_list<MeshBuilder> meshes )
+{
+  std::vector<mfem::Mesh*> mesh_list;
+  mesh_list.reserve( meshes.size() );
+  for ( auto& mesh : meshes ) {
+    // NOTE: the const cast is because the constructor requires a non-const mesh, even though the data is copied and not
+    // altered
+    mesh_list.push_back( &const_cast<mfem::Mesh&>( *mesh ) );
+  }
+  return mfem::Mesh( mesh_list.data(), mesh_list.size() );
+}
+
+MeshBuilder MeshBuilder::SquareMesh( int n_x_els, int n_y_els )
+{
+  return mfem::Mesh::MakeCartesian2D( n_x_els, n_y_els, mfem::Element::QUADRILATERAL );
+}
+
+MeshBuilder MeshBuilder::CubeMesh( int n_x_els, int n_y_els, int n_z_els )
+{
+  return mfem::Mesh::MakeCartesian3D( n_x_els, n_y_els, n_z_els, mfem::Element::HEXAHEDRON );
+}
+
+MeshBuilder MeshBuilder::HypercubeMesh( int dim, int n_els )
```

Comment:
It looks like we don't really need this HypercubeMesh() especially since Tribol just supports 2D and 3D, which here just calls MakeCartesian2D or 3D, which is done above in SquareMesh() or CubeMesh().